src/bittrex/api.rs

```diff
@@ -0,0 +1,157 @@
+//! Use this module to interact with the raw-original API provided by Bittrex.
+//! WARNING: Special attention should be paid to error management: parsing number, etc.
+
+#![allow(too_many_arguments)]
+
+use hmac::{Hmac, Mac};
+use sha2::{Sha256, Sha512, Digest};
+
+use hyper_native_tls::NativeTlsClient;
+use hyper::Client;
+use hyper::header;
+use hyper::net::HttpsConnector;
+
+use data_encoding::BASE64;
+
+use serde_json::Value;
+use serde_json::value::Map;
+
+use std::collections::HashMap;
+use std::io::Read;
+use std::thread;
+use std::time::Duration;
+use std::str;
+
+use error::*;
+use helpers;
+
+use exchange::Exchange;
+use coinnect::Credentials;
+use bittrex::utils;
+
+header! {
+    #[doc(hidden)]
+    (ApiSign, "apisign") => [String]
+}
+
+#[derive(Debug)]
+pub struct BittrexApi {
+    last_request: i64, // unix timestamp in ms, to avoid ban
+    api_key: String,
+    api_secret: String,
+    http_client: Client,
+    burst: bool,
+}
+
+
+impl BittrexApi {
+    /// Create a new BittrexApi by providing an API key & API secret
+    pub fn new<C: Credentials>(creds: C) -> Result<BittrexApi> {
+        if creds.exchange() != Exchange::Bittrex {
+            return Err(ErrorKind::InvalidConfigType(Exchange::Bittrex, creds.exchange()).into());
+        }
+
+        // TODO: implement correctly the TLS error in error_chain.
+        let ssl = match NativeTlsClient::new() {
+            Ok(res) => res,
+            Err(_) => return Err(ErrorKind::TlsError.into()),
+        };
+        let connector = HttpsConnector::new(ssl);
+
+        Ok(BittrexApi {
+               last_request: 0,
+               api_key: creds.get("api_key").unwrap_or_default(),
```

Comment:
I am just thinking here - what about `?`, what about invalid keys?

Reply:
That could be nice. I'll wait finishing Bittrex implementation and next I'll be looking at how to improve security and optimization

Reply:
@hugues31 I would like to help you, maybe you could create issue for that?

Reply:
@crackcomm Sure!